flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/mailbox/MailboxProcessor.java

```diff
@@ -179,13 +179,20 @@ public void allActionsCompleted() {
 			// keep state check and poison mail enqueuing atomic, such that no intermediate #close may cause a
 			// MailboxStateException in #sendPriorityMail.
 			if (mailbox.getState() == TaskMailbox.State.OPEN) {
-				sendPriorityMail(() -> mailboxLoopRunning = false, "poison mail");
+				sendControlMail(() -> mailboxLoopRunning = false, "poison mail");
 			}
 		});
 	}
 
-	private void sendPriorityMail(RunnableWithException priorityMail, String descriptionFormat, Object... descriptionArgs) {
-		mainMailboxExecutor.executeFirst(priorityMail, descriptionFormat, descriptionArgs);
+	/**
+	 * Sends the given <code>mail</code> using {@link TaskMailbox#putFirst(Mail)} .
+	 * Intended use is to control this <code>MailboxProcessor</code>; no interaction with tasks should be performed;
+	 */
+	private void sendControlMail(RunnableWithException mail, String descriptionFormat, Object... descriptionArgs) {
+		mailbox.putFirst(new Mail(mail,
```

Comment:
nit: the same formatting issue: `mail` should be in a new line as well, and the parameters should have a single indentation (or two for a function declaration)